portals/publisher/src/main/webapp/source/src/app/components/Apis/Create/GraphQL/Steps/ProvideGraphQL.jsx

```diff
@@ -106,31 +113,170 @@ export default function ProvideGraphQL(props) {
+        inputsDispatcher({ action: 'inputValue', value: null });
+        inputsDispatcher({ action: 'isFormValid', value: false });
+        inputsDispatcher({ action: 'endpoint', value: '' });
+    }
+
+    const isInvalidURL = Boolean(isValid.url);
+    let urlStateEndAdornment = null;
+    if (isValidating) {
+        urlStateEndAdornment = (
+            <InputAdornment position='end'>
+                <CircularProgress />
+            </InputAdornment>
+        );
+    } else if (isValid.url !== undefined) {
+        if (isInvalidURL) {
+            urlStateEndAdornment = (
+                <InputAdornment position='end'>
+                    <ErrorOutlineIcon fontSize='large' color='error' />
+                </InputAdornment>
+            );
+        } else {
+            urlStateEndAdornment = (
+                <InputAdornment position='end' id='url-validated'>
+                    <CheckIcon fontSize='large' color='primary' />
+                </InputAdornment>
+            );
+        }
+    }
+
+    const debouncedValidateURLOrEndpoint = useCallback(
+        debounce((newURL) => {
+            const handleResponse = (response) => {
+                const {
+                    body: { graphQLInfo },
+                } = response;
+                const isValidURL = response.body.isValid;
+                if (isValidURL) {
+                    inputsDispatcher({ action: 'graphQLInfo', value: graphQLInfo });
+                    setValidity({ isValidURL, file: null });
+                } else {
+                    let errorMessage;
+                    if (inputType === ProvideGraphQL.INPUT_TYPES.ENDPOINT) {
+                        errorMessage = 'Error occurred while generating GraphQL schema from endpoint';
+                    } else if (inputType === ProvideGraphQL.INPUT_TYPES.URL) {
+                        errorMessage = 'Error occurred while retrieving GraphQL schema from url';
+                    }
+                    setValidity({
+                        isValidURL,
+                        file: { message: errorMessage }
+                    });
+                }
+                onValidate(isValidURL);
+                setIsValidating(false);
+            };
+
+            const handleError = (error) => {
+                setValidity({ url: { message: error.message } });
+                onValidate(false);
+                setIsValidating(false);
+                console.error(error);
+            }
+            
+            const validationFunction = inputType === ProvideGraphQL.INPUT_TYPES.URL
+                ? () => API.validateGraphQL(newURL)
+                : () => API.validateGraphQL(newURL, { useIntrospection: true });
+
+            validationFunction()
+                .then(handleResponse)
+                .catch(handleError);
+
+            if(inputType === ProvideGraphQL.INPUT_TYPES.ENDPOINT){
+                inputsDispatcher({ action: 'endpoint', value: newURL });
+            }
+        }, 750),
```

Comment:
Remove hardcoded value and make it as constant

Reply:
Resolved via https://github.com/wso2/apim-apps/pull/1007